lib/active_merchant/billing/gateways/decidir.rb

```diff
@@ -0,0 +1,220 @@
+        post[:email] = options[:email] if options[:email]
+        post[:sub_payments] = []
+
+        add_invoice(post, money, options)
+        add_payment(post, credit_card, options)
+      end
+
+      def add_invoice(post, money, options)
+        post[:amount] = amount(money).to_i
+        post[:currency] = options[:currency] || currency(money)
+      end
+
+      def add_payment(post, credit_card, options)
+        card_data = {}
+        card_data[:card_number] = credit_card.number
+        card_data[:card_expiration_month] = format(credit_card.month, :two_digits)
+        card_data[:card_expiration_year] = format(credit_card.year, :two_digits)
+        card_data[:security_code] = credit_card.verification_value if credit_card.verification_value?
+        card_data[:card_holder_name] = credit_card.name if credit_card.name
+
+        # additional data used for Visa transactions
+        card_data[:card_holder_door_number] = options[:card_holder_door_number].to_i if options[:card_holder_door_number]
+        card_data[:card_holder_birthday] = options[:card_holder_birthday] if options[:card_holder_birthday]
+
+        card_data[:card_holder_identification] = {}
+        card_data[:card_holder_identification][:type] = options[:card_holder_identification_type] if options[:card_holder_identification_type]
+        card_data[:card_holder_identification][:number] = options[:card_holder_identification_number] if options[:card_holder_identification_number]
+
+        post[:card_data] = card_data
+      end
+
+      def headers(options = {})
+        {
+          'apikey' => @options[:api_key],
+          'Content-type'  => 'application/json',
+          'Cache-Control' => 'no-cache'
+        }
+      end
+
+      def commit(method, endpoint, parameters, options={})
+        url = "#{(test? ? test_url : live_url)}/#{endpoint}"
+
+        begin
+          raw_response = ssl_request(method, url, post_data(parameters), headers(options))
+          response = parse(raw_response)
+        rescue ResponseError => e
+          raw_response = e.response.body
+          response = parse(raw_response)
+        end
+
+        success = success_from(response)
+        Response.new(
+          success,
+          message_from(success, response),
+          response,
+          authorization: authorization_from(response),
+          test: test?,
+          error_code: success ? nil : error_code_from(response)
+        )
+      end
+
+      def post_data(parameters = {})
+        parameters.to_json
+      end
+
+      def parse(body)
+        JSON.parse(body)
+      rescue JSON::ParserError
+        {
+          'message' => "A non-JSON response was received from Decidir where one was expected. The raw response was:\n\n#{body}"
+        }
+      end
+
+      def message_from(success, response)
+        return response['status'] if success
+        return response['message'] if response['message']
+
+        msg = nil
```

Comment:
Nitpick: not a fan of abrvs (though I sympathize with the probable intent of distinguishing it from the response fields)

Reply:
😄I waffled on the abbreviation - started with `msg`, changed to `message` and then back to `msg` after seeing the abbreviation elsewhere in AM. Won't hurt my feelings to update it.